docs/api-process/check-out.md

```diff
@@ -1,6 +1,349 @@
+
+  - `token` - similarly to `id` this is a unique identifier, but it is also a public token. It will save the user's checkout session if, for example, they accidentally close the browser
+
+  - `totalPrice` - the total price of the checkout lines and shipping costs
+
+  - `availablePaymentGateways` - a list of payment gateways which are currently configured on your Saleor server and can be used to pay for the checkout.
+
+  - `availableShippingMethods` - a list of available shipping methods for this checkout. If the items in the cart require shipment, setting a shipping method is mandatory.
+
+- `created` - a boolean field which tells whether a new checkout object was created, or an existing one was used. E.g., if an authenticated user had a unfinished checkout before, it would be reused.
+
+- `errors` - list of errors that could occur during mutation execution
+
+The following example shows how the `checkoutCreate` mutation creates the `Checkout` object and returns the checkout information:
+
+```graphql
+mutation {
+  checkoutCreate(
+    input: {
+      email: "customer@example.com"
+      lines: [{ quantity: 1, variantId: "UHJvZHVjdFZhcmlhbnQ6Mjk3" }]
+      shippingAddress: {
+        firstName: "John"
+        lastName: "Doe"
+        streetAddress1: "1470  Pinewood Avenue"
+        city: "Michigan"
+        postalCode: "49855"
+        country: "US"
+        countryArea: "MI"
+      }
+      billingAddress: {
+        firstName: "John"
+        lastName: "Doe"
+        streetAddress1: "1470  Pinewood Avenue"
+        city: "Michigan"
+        postalCode: "49855"
+        country: "US"
+        countryArea: "MI"
+      }
+    }
+  ) {
+    checkout {
+      id
+      totalPrice {
+        gross {
+          amount
+          currency
+        }
+      }
+      availableShippingMethods {
+        id
+        name
+      }
+      availablePaymentGateways
+    }
+    errors {
+      field
+      message
+    }
+  }
+}
+```
+
+As a result we're getting a newly created checkout object for which we return its ID, total price, list of available shipping and payment methods:
+
+```json
+{
+  "data": {
+    "checkoutCreate": {
+      "checkout": {
+        "id": "Q2hlY2tvdXQ6ZmE5ZjBkMjYtMWM3NC00MDgyLTk3MzktYTIxOGE2NzVjMDZk",
+        "totalPrice": {
+          "gross": {
+            "amount": 20,
+            "currency": "USD"
+          }
+        },
+        "availableShippingMethods": [
+          {
+            "id": "U2hpcHBpbmdNZXRob2Q6MTM=",
+            "name": "UPS"
+          },
+          {
+            "id": "U2hpcHBpbmdNZXRob2Q6MTI=",
+            "name": "DHL"
+          }
+        ],
+        "availablePaymentGateways": ["BRAINTREE"]
+      },
+      "errors": []
+    }
+  }
+}
+```
+
+## Step #2 (optional) - `checkoutShippingMethodUpdate`
+
+This step is only used if purchased items require shipping (if they are physical products). The user must select a specific shipping method to create shipping for this checkout.
```

Comment:
Do we have a way to check if checkout _needs_ shipping?

Reply:
Yup! `checkout.isShippingRequired`

Reply:
But you have to create the checkout object first, we'll eventually get to documenting how you can add addresses later.

Reply:
Would be awesome if we mention it here